src/iter/filter_map.rs

```diff
@@ -59,25 +59,25 @@ impl<M, FILTER_OP, R> BoundedParallelIterator for FilterMap<M, FILTER_OP>
 /// ////////////////////////////////////////////////////////////////////////
 /// Consumer implementation
 
-struct FilterMapConsumer<'f, C, FILTER_OP: 'f> {
+struct FilterMapConsumer<'p, C, P: 'p> {
     base: C,
-    filter_op: &'f FILTER_OP,
+    filter_op: &'p P,
 }
 
-impl<'f, C, FILTER_OP: 'f> FilterMapConsumer<'f, C, FILTER_OP> {
-    fn new(base: C, filter_op: &'f FILTER_OP) -> Self {
+impl<'p, C, P: 'p> FilterMapConsumer<'p, C, P> {
+    fn new(base: C, filter_op: &'p P) -> Self {
         FilterMapConsumer {
             base: base,
             filter_op: filter_op,
         }
     }
 }
 
-impl<'f, ITEM, MAPPED_ITEM, C, FILTER_OP> Consumer<ITEM> for FilterMapConsumer<'f, C, FILTER_OP>
-    where C: Consumer<MAPPED_ITEM>,
-          FILTER_OP: Fn(ITEM) -> Option<MAPPED_ITEM> + Sync + 'f
+impl<'p, I, MI, C, P> Consumer<I> for FilterMapConsumer<'p, C, P>
```

Comment:
If we were using `T` for `Item`, I *think* I'd prefer `U` for `MAPPED_ITEM`. If we're going into single letter territory, might as well *go there*, I guess. Somewhat oddly (to my mind), the standard library [seems to adopt `B` for this purpose](https://doc.rust-lang.org/std/iter/struct.Map.html), though.